mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/KeyView.m

```diff
@@ -60,22 +63,26 @@ - (id)initWithFrame:(NSRect)frame {
         [_label setLineBreakMode:NSLineBreakByClipping];
         [self addSubview:_label];
 
-        bgColor1 = [self getOpaqueColorWithRed:209 green:211 blue:212];
-        bgColor2 = [self getOpaqueColorWithRed:166 green:169 blue:172];
+        bgColorRegularKey = [self getOpaqueColorWithRed:209 green:211 blue:212];
+        bgColorSpecialKey = [self getOpaqueColorWithRed:166 green:169 blue:172];
     }
 
     return self;
 }
 
 - (void)drawRect:(NSRect)rect {
+    os_log_t oskKeyLog = os_log_create("org.sil.keyman", "osk-key");
+    os_log_with_type(oskKeyLog, OS_LOG_TYPE_DEBUG, "KeyView drawRect: %{public}@, bounds: %{public}@, keyCode: 0x%lx, caption: %{public}@, label: %{public}@", NSStringFromRect(rect), NSStringFromRect(self.bounds), self.keyCode, self.caption.stringValue, self.label.stringValue);
```

Comment:
More "new logs".  This one looks like we're making a new logging object... with the same ID as the log noted for the previous comment?

Reply:
The os_log_create does not actually create a new log object if it has already been called. Internally, the logging framework creates a singleton for each log object of a certain subsystem/category.

To reduce clutter, I can create a private member variable to hold this reference so that within each method there is only one line of code to write a log statement. It will have no effect on performance, but could make it easier to read the code.